src/detect.ts

```diff
@@ -61,6 +63,20 @@ export function detectSync(options: DetectOptions = {}): DetectResult | null {
   return null
 }
 
+/**
+ * Detects the package manager used in the running process.
+ *
+ * This method will check for `process.env.npm_config_user_agent`.
+ */
+export function getUserAgent(): AgentName | null {
+  const userAgent = process.env.npm_config_user_agent
+  if (!userAgent)
+    return null
+
+  const name = userAgent.split(' ')[0].split('/')[0] as AgentName
```

Comment:
I don't think we first need to split on empty space (`' '`) and we can just split on `/` directly, just like we do in the Remix codebase?
https://github.com/remix-run/remix/blob/8649f1a561ce5ea0bfba5ab862c0c94052f4a5c3/packages/remix-dev/cli/detectPackageManager.ts#L10-L23

```suggestion
  const name = userAgent.split('/')[0] as AgentName
```